### PR TITLE
Add temporary "create_package" target to cmake and remove scripts to do the same

### PR DESCRIPTION
**Pull request by marco-m**
Usage: `make create_package`. Will build QSyncthingTray if needed.

- Tested on Mac.
- Added also support for Windows but it requires you to fill the missing parameters to the invocation of windeployqt and see if it works.

Temporary because the proper way is to use CPack, but I want to go there little by little :-)

### CODE REVIEW

**Comment by sieren** [comment]
Looks great! I could add those parameters.. or if you wanna amend them, they are `--compiler-runtime` -- i'd merge this then 👍

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,3 +118,38 @@ else()
...
+      NAMES macdeployqt
+      PATHS ${qt5_bin_dir})
+  if(NOT qt5_qtdeploy)
+    message(FATAL_ERROR "macdeployqt not found")
+  endif()
+
+  add_custom_target(create_package
+      COMMAND ${CMAKE_COMMAND} -E remove_directory QSyncthingTray.app
+      COMMAND ${CMAKE_COMMAND} -E remove QSyncthingTray.dmg
+      #COMMAND ${CMAKE_COMMAND} --build . --target clean --config $<CONFIG>
```
Comment (sieren) [inline CMakeLists.txt]: Also I guess this isn't needed?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.2)
 
-project (QSyncthingTray)
+project (QSyncthingTray LANGUAGES CXX)
```
Comment (marco-m) [inline CMakeLists.txt]: This is for speed. It skips C and goes slightly faster when configuring.

File: CMakeLists.txt (hunk 1 above)
Comment (marco-m) [inline CMakeLists.txt]: yes, it is not needed. I will remove it.

**Comment by marco-m** [comment]
Updated PR addressing comments

**Comment by sieren** [comment]
Cool! Thanks! :) I'll test this on my Windows box this weekend and then we can merge. Another good idea might be to test the packaging itself on CI.

**Comment by sieren** [comment]
Doesn't work on Windows right now (see comment).
On that note, how do I specify the build config using cmake (Debug/Release) ?

**Comment by marco-m** [comment]
What "doesn't work on Windows" ? Running "make create_package" ? Could you please add the cmake command-line you use to configure the project and the error output?

**Comment by sieren** [comment]
I don't have `make` on my Windows/Git Bash environment (I guess I could add it by using CygWin). Even if it had `make` (I'm using `msbuild` on the `.vcproj` files) this would fail due to the missing `.exe` behind `QSyncthingTray` (https://github.com/sieren/QSyncthingTray/pull/204#discussion_r99486422).

**Comment by sieren** [comment]
```
$ cmake ../ -DCMAKE_PREFIX_PATH=/c/Qt/5.7/msvc2015_64/ -G "Visual Studio 14 2015 Win64"
[...]
$ /c/Program\ Files\ \(x86\)/MSBuild/14.0/Bin/MSBuild.exe create_package.vcxproj
[...]
  Done Building Project "C:\Users\sierenmusic\Development\QSyncthingTray\build\ALL_BUILD.vcxproj" (default targets).

  Build succeeded.

  "C:\Users\sierenmusic\Development\QSyncthingTray\build\ALL_BUILD.vcxproj" (default target) (1) ->
  "C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj" (default target) (2) ->
  (ClCompile target) ->
cl : Command line warning D9002: ignoring unknown option '-std=c++11' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
C:\Users\sierenmusic\Development\QSyncthingTray\sources\qst\syncconnector.cpp(420): warning C4101: 'e': unreferenced local variable [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
C:\Users\sierenmusic\Development\QSyncthingTray\sources\qst\window.cpp(297): warning C4101: 'e': unreferenced local variable [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2230): warning C4996: 'std::_Copy_impl': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2507): warning C4996: 'std::_Copy_backward': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]

      5 Warning(s)
      0 Error(s)

  Time Elapsed 00:01:17.46
  "QSyncthingTray" does not exist.

C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 1. [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
Done Building Project "C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj" (default targets) -- FAILED.

Build FAILED.

"C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj" (default target) (1) ->
(CustomBuild target) ->
  cl : Command line warning D9002: ignoring unknown option '-std=c++11' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Users\sierenmusic\Development\QSyncthingTray\sources\qst\syncconnector.cpp(420): warning C4101: 'e': unreferenced local variable [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Users\sierenmusic\Development\QSyncthingTray\sources\qst\window.cpp(297): warning C4101: 'e': unreferenced local variable [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2230): warning C4996: 'std::_Copy_impl': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2507): warning C4996: 'std::_Copy_backward': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  cl : Command line warning D9002: ignoring unknown option '-std=c++11' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Users\sierenmusic\Development\QSyncthingTray\sources\qst\syncconnector.cpp(420): warning C4101: 'e': unreferenced local variable [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Users\sierenmusic\Development\QSyncthingTray\sources\qst\window.cpp(297): warning C4101: 'e': unreferenced local variable [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2230): warning C4996: 'std::_Copy_impl': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2507): warning C4996: 'std::_Copy_backward': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' [C:\Users\sierenmusic\Development\QSyncthingTray\build\QSyncthingTray.vcxproj] [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]


"C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj" (default target) (1) ->
(CustomBuild target) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 1. [C:\Users\sierenmusic\Development\QSyncthingTray\build\create_package.vcxproj]

    10 Warning(s)
    1 Error(s)

Time Elapsed 00:01:19.73

**Comment by sieren** [comment]
https://github.com/sieren/QSyncthingTray/pull/204#discussion_r99486422 fixes this, but the Debug build issue remains

**Comment by sieren** [comment]
Update: Debug issue is actually not an issue, that was on me and my lack of knowledge about MSBuild. Figured it out though.

**Comment by marco-m** [comment]
Hello, I am confused by the link you added in "So once #204 (comment) is added we can merge this" because it points at the beginning of _this_ same PR (nor at a specific comment), so I don't know how to proceed :-)

**Comment by sieren** [comment]
@marco-m opps.. didnt hit "Submit Review"... :)

**Comment by marco-m** [comment]
Addressed review comments.

**Comment by sieren** [comment]
👍 Merged

**Pull request merged**